include/openenclave/seal.h

```diff
@@ -0,0 +1,299 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+/**
+ * @file
+ *
+ * This file defines constants and structures for sealing APIs.
+ *
+ * Only TEE agnostic definitions should go in this file. SGX specific
+ * definitions should go in sgx/seal.h
+ *
+ */
+#ifndef _OE_SEAL_H
+#define _OE_SEAL_H
+
+#include <openenclave/bits/evidence.h>
+#include <openenclave/bits/result.h>
+#include <openenclave/bits/types.h>
+
+OE_EXTERNC_BEGIN
+
+/**
+ * Seal settings as TLV tuples.
+ *
+ * It's strongly recommended to use OE_SEAL_SET_* helper macros to set up this
+ * structure, rather than assigning to members directly.
+ */
+typedef struct _oe_seal_setting
+{
+    int type;      ///< Setting type. See oe_seal_setting_type_t for details.
```

Comment:
```suggestion
    oe_seal_setting_type_t type;      ///< Setting type.
```

Reply:
We also discussed this in sig-arch last week. `type` can take not only values defined by OE, but also values defined by plug-ins. Thus, we cannot use `oe_seal_setting_type_t` as its type.